src/sys/signal.rs

```diff
@@ -388,8 +388,12 @@ pub fn pthread_sigmask(how: SigFlags,
     Errno::result(res).map(drop)
 }
 
-pub fn kill(pid: libc::pid_t, signal: Signal) -> Result<()> {
-    let res = unsafe { libc::kill(pid, signal as libc::c_int) };
+pub fn kill(pid: libc::pid_t, signal: Option<Signal>) -> Result<()> {
```

Comment:
maybe we can take advantage of the [recently added](https://github.com/rust-lang/rust/pull/34828) `impl From<T> for Option<T>`. On recent enough Rust (1.12+), `kill(pid, SIGBLAH)` will still work, so it's a bit more ergonomic.